### PR TITLE
feat: add styling and some updates for pick-list nesting

### PR DESCRIPTION
**Pull request by asangma**
**Related Issue:** #767

### CODE REVIEW

File: src/components/calcite-pick-list-group/calcite-pick-list-group.tsx [hunk 1]
```diff
@@ -37,14 +39,19 @@ export class CalcitePickListGroup {
...
+    };
+
     return (
       <Host>
-        {this.textGroupTitle ? <h3 class={CSS.heading}>{this.textGroupTitle}</h3> : null}
-        <slot name={"parent-item"} />
-        <section class={classes}>
+        {textGroupTitle ? <h3 class={CSS.heading}>{textGroupTitle}</h3> : null}
+        <slot name={SLOTS.parentItem} />
+        <section class={classnames(sectionClasses)}>
```
Comment (driskull) [inline src/components/calcite-pick-list-group/calcite-pick-list-group.tsx]: nitpick: classnames isn't needed here.

File: src/components/calcite-pick-list-item/calcite-pick-list-item.tsx [hunk 2]
```diff
@@ -207,6 +207,15 @@ export class CalcitePickListItem {
     );
   }
 
+  renderSecondaryAction() {
+    const hasSecondaryAction = this.el.querySelector(`[slot=${SLOTS.secondaryAction}]`);
```
Comment (driskull) [inline src/components/calcite-pick-list-item/calcite-pick-list-item.tsx]: `this.el.querySelector(`[slot=${SLOTS.secondaryAction}]`)` is very common now it seems. Maybe we should have a utility for this @jcfranco? `hasSlot(el: HTMLElement, slotName:string): boolean;`?

Comment (jcfranco) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.tsx]: That can be arranged! 😄

Comment (driskull) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.tsx]: https://github.com/Esri/calcite-app-components/issues/784

File: src/components/calcite-pick-list-group/calcite-pick-list-group.tsx (hunk 1 above)
Comment (asangma) [reply src/components/calcite-pick-list-group/calcite-pick-list-group.tsx]: Oh dang!

File: src/components/calcite-pick-list-item/calcite-pick-list-item.tsx (hunk 2 above)
Comment (asangma) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.tsx]: Noice!

File: src/components/calcite-pick-list-group/calcite-pick-list-group.tsx (hunk 1 above)
Comment (asangma) [reply src/components/calcite-pick-list-group/calcite-pick-list-group.tsx]: Updated! @driskull

**Comment by driskull** [comment]
Murge?

**Comment by asangma** [comment]
@driskull  If @pr3tori4n approves.

File: src/components/calcite-pick-list-group/calcite-pick-list-group.scss [hunk 3]
```diff
@@ -14,16 +14,18 @@
 
 h3.heading {
   font-size: var(--calcite-app-font-size-0);
-  font-weight: var(--calcite-app-font-weight);
-  margin: var(--calcite-app-cap-spacing-half) var(--calcite-app-side-spacing-half);
+  margin: var(--calcite-app-cap-spacing-half) var(--calcite-app-side-spacing);
 }
 
 .container {
```
Comment (pr3tori4n) [inline src/components/calcite-pick-list-group/calcite-pick-list-group.scss]: is this container element still needed?

**Comment by asangma** [comment]
@pr3tori4n Gonna merge! Thanks man.

**Pull request merged**